src/coreComponents/common/logger/ErrorHandling.hpp

```diff
@@ -0,0 +1,254 @@
+ * Copyright (c) 2023-2024 Chevron
+ * Copyright (c) 2019-     GEOS/GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file ErrorHandling.hpp
+ */
+
+#ifndef INITIALIZATION_ERROR_LOGGER_HPP
+#define INITIALIZATION_ERROR_LOGGER_HPP
+
+#include "common/DataTypes.hpp"
+
+namespace geos
+{
+
+/**
+ * @class ErrorLogger
+ * @brief Class to format and write the error/warning message that occured during the initialization
+ */
+class ErrorLogger
+{
+
+public:
+
+  /**
+   * @brief Construct a new Error Logger object
+   *
+   */
+  ErrorLogger();
+
+  /**
+   * @brief Create the yaml file if the option is specified in the command line options
+   * 
+   */
+  void createFile();
+
+  /**
+   * @enum MsgType
+   * Enum listing the different types of possible errors
+   */
+  enum class MsgType
+  {
+    Error,
+    Warning,
+    Exception
+  };
+
+  /**
+   * @brief Stores contextual information about the error that occurred and assigns it a priority (default is 0)
+   * 
+   */
+  struct ContextInfo
+  {
+    map< std::string, std::string > m_ctxInfo;
+    integer m_priority = 0;
+
+    /**
+     * @brief Set the priority of the current error context information
+     * 
+     * @param priority 
+     * @return ContextInfo& 
+     */
+    ContextInfo & setPriority( integer priority )
+    { m_priority = priority; return *this; }
+  };
+
+  /**
+   * @brief Struct to define the error/warning message
+   *
+   */
+  struct ErrorMsg
+  {
+    MsgType m_type;
+    std::string m_msg;
+    std::string m_file;
+    integer m_line;
+    std::vector< int > m_ranksInfo;
+    std::vector< ContextInfo > m_contextsInfo;
+    std::vector< std::string > m_sourceCallStack;
+
+    int n = 0;
+
+    /**
+     * @brief Construct a new Error Msg object
+     *
+     */
+    ErrorMsg() {};
+
+    /**
+     * @brief Construct a new Error Msg object
+     *
+     * @param msgType The type of the message (error or warning)
+     * @param msgContent The error/warning message content
+     * @param msgFile The file name where the error occcured
+     * @param msgLine The line where the error occured
+     */
+    ErrorMsg( MsgType msgType, std::string msgContent, std::string msgFile, integer msgLine )
+      : m_type( msgType ), m_msg( msgContent ), m_file( msgFile ), m_line( msgLine ) {}
+
+
+    ErrorLogger * parent = nullptr;
+
+    /**
+     * @brief Fill the msg field of the structure with the error message
+     *
+     * @param e is the exception
+     * @return ErrorMsg&
+     */
+    ErrorMsg & addToMsg( std::exception const & e );
+
+    /**
+     * @brief
+     *
+     * @param msg Add information about the error that occured to the msg field of the structure
+     * @return ErrorMsg&
+     */
+    ErrorMsg & addToMsg( std::string const & msg );
```

Comment:
1. Try to fill every comment lines,
2. In modern C++, a general rule is to prefer `string_view` over `string` (=copy), `string const &` and `char const *` when giving a string as parameter,
3. I'm proposing a rewritten version of the description, If you want it, you can use it on the other signature of `addToMsg()`,
4. No empty comment lines.

```suggestion
    /**
     * @brief Add text to the error msg that occured to the msg field of the structure
     * @param msg The text to add.
     * @return The instance, for builder pattern.
     */
    ErrorMsg & addToMsg( std::string const & msg );
```

If you are ok with those comments, please impact the other comments of this header.

Reply:
If I change msg to a std::string_view I have a type problem in the following implementation :   `parent->m_currentErrorMsg.m_msg = errorMsg + parent->m_currentErrorMsg.m_msg; `

Reply:
In header, string_view are prefered to not couple every text parameter to std::string type.

Reply:
I applied the solution we saw together.